### PR TITLE
Add check so that people can’t register if "anonymous" is even part o…

### LINKED ISSUE #39: Add feature so that people can’t register with anonymous as a username
Estimated effort: M t-shirt size
Actual effort: S t-shirt size

Milestones: Sprint 2
Acceptance criteria: Ensure a new user cannot register with "Anonymous" or a variation as username

### PR DESCRIPTION
**Pull request by gekiclaws**

### CODE REVIEW

**Comment by margarethe913** [comment]
I think there are lint errors to fix!

File: nodebb-theme-mytheme/templates/quickreply.tpl [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{{ if privileges.topics:reply }}}
+<div component="topic/quickreply/container" class="quick-reply d-flex gap-3 mb-4">
+	<div class="icon hidden-xs">
+		<a class="d-inline-block position-relative" href="{{{ if loggedInUser.userslug }}}{config.relative_path}/user/{loggedInUser.userslug}{{{ else }}}#{{{ end }}}">
```
Comment (margarethe913) [inline nodebb-theme-mytheme/templates/quickreply.tpl:4]: are these changes from a different commit?

File: public/src/client/topic/postTools.js [hunk 2]
```diff
@@ -125,6 +125,30 @@ define('forum/topic/postTools', [
 			});
 		});
 
+		postContainer.on('click', '.composer-submit-anonymous', function (e) {
```
Comment (margarethe913) [inline public/src/client/topic/postTools.js:128]: lint errors + are these from this commit?

File: src/plugins/nodebb-plugin-composer-default/.jshintrc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "bitwise" : true, // true: Prohibit bitwise operators (&, |, ^, etc.)
+    "camelcase" : false, // true: Identifiers must be in camelCase
+    "curly" : true, // true: Require {} for every new block or scope
+    "eqeqeq" : true, // true: Require triple equals (===) for comparison
+    "forin" : true, // true: Require filtering for..in loops with obj.hasOwnProperty()
+    "immed" : false, // true: Require immediate invocations to be wrapped in parens e.g. `(function () { } ());`
+    "indent" : 4, // {int} Number of spaces to use for indentation
+    "latedef" : false, // true: Require variables/functions to be defined before being used
+    "newcap" : false, // true: Require capitalization of all constructor functions e.g. `new F()`
+    "noarg" : true, // true: Prohibit use of `arguments.caller` and `arguments.callee`
```
Comment (margarethe913) [inline src/plugins/nodebb-plugin-composer-default/.jshintrc:17]: Should this file be in the PR?

**Review by margarethe913 — CHANGES_REQUESTED** [review]
i think there are duplicate commits/changes in this PR + lint errors

File: nodebb-theme-mytheme/templates/quickreply.tpl (hunk 1 above)
Comment (gekiclaws) [reply nodebb-theme-mytheme/templates/quickreply.tpl:4]: yep they're not from my commit.

File: public/src/client/topic/postTools.js (hunk 2 above)
Comment (gekiclaws) [reply public/src/client/topic/postTools.js:128]: yep they're not from my commit.

**Pull request closed without merging**